crates/runtime/src/embeddings/vector_search.rs

```diff
@@ -345,7 +345,13 @@ impl VectorSearchTableResult {
         let result = embedding_records
             .iter()
             .flat_map(|v| {
-                if let Some(col) = v.column(0).as_any().downcast_ref::<StringArray>() {
+                let embedded_column = v.column(0);
+                if let Some(col) = embedded_column.as_any().downcast_ref::<StringArray>() {
+                    col.iter()
+                        .map(|v| v.unwrap_or_default().to_string())
+                        .collect::<Vec<String>>()
+                } else if let Some(col) = embedded_column.as_any().downcast_ref::<StringViewArray>()
+                {
```

Comment:
Weird styling. No prettier?

Reply:
Going to merge this as-is and review how to refactor this- this pattern is used in several places in vector search module. 

```
col.iter()
  .map(|v| v.unwrap_or_default().to_string())
  .collect::<Vec<String>>()
```

I also see that there are places where we default to empty vec instead of failling / reporting an error due to unexpected search result - will be improved.

```
if let Some(col) = v.as_any().downcast_ref::<arrow::array::Float64Array>() {
      col.iter().collect::<Vec<Option<f64>>>()
  } else {
      vec![]
  }
```